ee/hogai/graph/billing/nodes.py

```diff
@@ -0,0 +1,205 @@
+        return template.format_prompt(**template_data).to_string()
+
+    def _format_usage_history_table(self, usage_history) -> str:
+        """Format multiple breakdowns as compact weekly breakdown table."""
+
+        # Create dynamic header with breakdown columns
+        breakdown_names = [result.breakdown_value or "Unknown" for result in usage_history]
+        header = "| Week | " + " | ".join(breakdown_names) + " |"
+        separator = "|------|" + "|".join(["-------"] * len(breakdown_names)) + "|"
+
+        table_lines = [header, separator]
+
+        # Get all unique weeks across all breakdowns
+        all_weeks: set[str] = set()
+        breakdown_data: dict[int, dict[str, float]] = {}
+        week_date_mapping: dict[str, str] = {}  # Maps week_key to actual date range
+
+        for i, result in enumerate(usage_history):
+            breakdown_data[i] = {}
+            if result.dates and result.data:
+                for date, usage in zip(result.dates, result.data):
+                    try:
+                        date_obj = datetime.strptime(date, "%Y-%m-%d")
+                        # Calculate the start of the week (Monday)
+                        days_since_monday = date_obj.weekday()
+                        week_start = date_obj - timedelta(days=days_since_monday)
+                        week_end = week_start + timedelta(days=6)
+                        week_key = week_start.strftime("%Y-%m-%d")  # Use start date as key
+                        week_range = f"{week_start.strftime('%b %d')} - {week_end.strftime('%b %d')}"
+
+                        all_weeks.add(week_key)
+                        week_date_mapping[week_key] = week_range
+                        breakdown_data[i][week_key] = float(usage) if usage is not None else 0
+                    except (ValueError, TypeError):
+                        # Fallback to original date format if parsing fails
+                        week_key = date
+                        all_weeks.add(week_key)
+                        week_date_mapping[week_key] = date
+                        breakdown_data[i][week_key] = float(usage) if usage is not None else 0
+
+        # Sort weeks and show most recent 4-6 weeks
+        sorted_weeks = sorted(all_weeks)[-6:] if len(all_weeks) > 6 else sorted(all_weeks)
+
+        # Create rows for each week
+        for week in sorted_weeks:
+            week_totals = []
+            for i in range(len(usage_history)):
+                total = breakdown_data[i].get(week, 0)
+                if total >= 1000:
+                    week_totals.append(f"{total:,.0f}")
+                else:
+                    week_totals.append(f"{total:.0f}")
+
+            week_display = week_date_mapping.get(week, week)
+            week_row = f"| {week_display} | " + " | ".join(week_totals) + " |"
+            table_lines.append(week_row)
+
+        return "\n".join(table_lines)
+
+    def _get_top_events_by_usage(self) -> list[dict[str, Any]]:
```

Comment:
This will be team scoped, and it's often very useful to provide customers with insights on which team (project) is driving most of their usage/spend. Usage/spend endpoints query data across teams but when running this query, it will be scoped to the team, so we might need to tell user sth like "looks like your product analytics usage is coming mainly from project B, while you're currently on project A, switch there and..." (not sure if Max convo will be there to pick up history or if they need to start a new one).